skimage/filters/rank/tests/test_rank.py

```diff
@@ -720,3 +691,19 @@ def test_majority(self):
         expected = rank.windowed_histogram(
             img, elem).argmax(-1).astype(np.uint8)
         assert_equal(expected, rank.majority(img, elem))
+
+    def test_output_same_dtype(self):
+        image = (np.random.rand(100, 100) * 256).astype(np.uint8)
+        out = np.empty_like(image)
+        mask = np.ones(image.shape, dtype=np.uint8)
+        elem = np.ones((3, 3), dtype=np.uint8)
+        rank.maximum(image=image, selem=elem, out=out, mask=mask)
+        assert_equal(image.dtype, out.dtype)
+
+    def test_input_boolean_dtype(self):
+        image = (np.random.rand(100, 100) * 256).astype(np.bool_)
+        out = np.empty_like(image)
+        mask = np.ones(image.shape, dtype=np.bool_)
+        elem = np.ones((3, 3), dtype=np.bool_)
+        # rank.maximum(image=image, selem=elem, out=out, mask=mask)
```

Comment:
WHy this is commented? (2 lines)

Reply:
This currently fails w/ a `TypeError: No matching signature found` in the `skimage/filters/rank/generic_cy.pyx` file. I suspect it's b/c I need to do something explicitly in the cython to support `np.bool_`, but can't figure out how to do that or if that is the actual cause.

Reply:
I have no idea on where to start for cython debugging, so any help is appreciated

Reply:
I have no idea on where to start for cython debugging, so any help is appreciated